state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -217,6 +213,51 @@ pub enum PalletConfigUpdate<T: Config<I>, I: 'static = ()> {
 	SetMaxSwapRetryDurationBlocks { blocks: BlockNumber },
 }
 
+macro_rules! append_chain_to_name {
+	($name:ident) => {
+		match T::TargetChain::NAME {
+			"Ethereum" => concat!(stringify!($name), "Ethereum"),
+			"Polkadot" => concat!(stringify!($name), "Polkadot"),
+			"Bitcoin" => concat!(stringify!($name), "Bitcoin"),
+			"Arbitrum" => concat!(stringify!($name), "Arbitrum"),
+			"Solana" => concat!(stringify!($name), "Solana"),
+			_ => concat!(stringify!($name), "Other"),
+		}
+	};
+}
+
+impl<T, I> TypeInfo for PalletConfigUpdate<T, I>
```

Comment:
How can we know that this works? There are no compiler guarantees of anything here right?

Reply:
no, the downside of implementing it manually is that we need to update the typeinfo together with the struct/enum itself.